datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -360,6 +366,14 @@ public void stop() {
       }
     }
 
+    // Shutdown executor services
+    if (_scheduledExecutor != null) {
+      _scheduledExecutor.shutdown();
```

Comment:
I'm calling the more graceful shutdown path here. No new submissions will be accepted but the tasks that have already been submitted will be lingering until completion. I could block and wait for all tasks to complete here. I could also force terminate everything (which is a best effort operation). I don't think more oppressive measures are required here.